File: 500/main.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	fmt.Println(findWords([]string{"Hello", "Alaska", "Dad", "Peace"}))
}

func findWords(words []string) []string {
	firstRow := "qwertyuiop"
	secondRow := "asdfghjkl"
	thirdRow := "zxcvbnm"
	result := []string{}
	for _, word := range words {
		lowerWord := strings.ToLower(word)
		if strings.IndexByte(firstRow, lowerWord[0]) != -1 {
			if inRow(lowerWord, firstRow) {
				result = append(result, word)
			}
		} else if strings.IndexByte(secondRow, lowerWord[0]) != -1 {
			if inRow(lowerWord, secondRow) {
				result = append(result, word)
			}
		} else if strings.IndexByte(thirdRow, lowerWord[0]) != -1 {
			if inRow(lowerWord, thirdRow) {
				result = append(result, word)
			}
		}
	}
	return result
}

func inRow(word string, row string) bool {
	for _, char := range word {
		if strings.IndexRune(row, char) == -1 {
			return false
		}
	}
	return true
}

File: 500/main.py
"""
Given a List of words, return the words that can be typed using letters of alphabet on only one row's of American keyboard like the image below.

American keyboard

Example 1:

Input: ["Hello", "Alaska", "Dad", "Peace"]
Output: ["Alaska", "Dad"]

Note:

    You may use one character in the keyboard more than once.
    You may assume the input string will only contain letters of alphabet.
"""

class Solution(object):
    def findWords(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        self.res = []
        for word in words:
            self.checkWord(word)
        return self.res

    def checkWord(self, word):
        rows = [
            'qwertyuiopQWERTYUIOP',
            'asdfghjklASDFGHJKL',
            'zxcvbnmZXCVBNM'
        ]
        for row in rows:
            if len(list(filter(lambda ch: ch in row, word))) == len(word):
                self.res.append(word)
                return

ans = Solution()
print(ans.findWords(["Hello", "Alaska", "Dad", "Peace"]))
